internal/tls/config.go

```diff
@@ -0,0 +1,134 @@
+package tls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type TLSConfig struct {
+	CertPath         string
+	KeyPath          string
+	RootCertPath     string
+	CAPath           string
+	Cipher           string
+	VerifyServerCert bool
+}
+
+func NewTLSConfig(certPath, keyPath, rootCertPath, caPath, cipher string, verifyServerCert bool) *TLSConfig {
+	return &TLSConfig{
+		CertPath:         certPath,
+		KeyPath:          keyPath,
+		RootCertPath:     rootCertPath,
+		CAPath:           caPath,
+		Cipher:           cipher,
+		VerifyServerCert: verifyServerCert,
+	}
+}
+
+func (c *TLSConfig) BuildTLSConfig() (*tls.Config, error) {
+	var rootCAs *x509.CertPool
+	var err error
+
+	if c.RootCertPath != "" || c.CAPath != "" {
+		rootCAs = x509.NewCertPool()
+
+		if c.RootCertPath != "" {
+			rootCert, err := os.ReadFile(c.RootCertPath)
+			if err != nil {
+				return nil, fmt.Errorf("failed to read SSL root certificate from %s: %w", c.RootCertPath, err)
+			}
+			if !rootCAs.AppendCertsFromPEM(rootCert) {
+				return nil, fmt.Errorf("failed to parse SSL root certificate from %s", c.RootCertPath)
+			}
+		}
+
+		if c.CAPath != "" && c.CAPath != c.RootCertPath {
+			caCert, err := os.ReadFile(c.CAPath)
+			if err != nil {
+				return nil, fmt.Errorf("failed to read SSL CA certificate from %s: %w", c.CAPath, err)
+			}
+			if !rootCAs.AppendCertsFromPEM(caCert) {
+				return nil, fmt.Errorf("failed to parse SSL CA certificate from %s", c.CAPath)
+			}
+		}
+	} else if c.VerifyServerCert {
+		rootCAs, err = x509.SystemCertPool()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get system cert pool: %w", err)
+		}
+	}
+
+	tlsConfig := &tls.Config{
+		RootCAs:            rootCAs,
+		InsecureSkipVerify: !c.VerifyServerCert,
+	}
+
+	if c.CertPath != "" && c.KeyPath != "" {
+		cert, err := tls.LoadX509KeyPair(c.CertPath, c.KeyPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load SSL certificate pair (cert: %s, key: %s): %w",
+				c.CertPath, c.KeyPath, err)
+		}
+		tlsConfig.Certificates = []tls.Certificate{cert}
+	}
+
+	if c.Cipher != "" {
+		cipherSuites := parseCipherSuites(c.Cipher)
+		if len(cipherSuites) > 0 {
+			tlsConfig.CipherSuites = cipherSuites
+		}
+	}
+
+	return tlsConfig, nil
+}
+
+// parseCipherSuites parses a colon-separated list of cipher suite names
+// and returns the corresponding cipher suite IDs
+func parseCipherSuites(cipherStr string) []uint16 {
```

Comment:
does this have any implication/impact I should be aware of in the context of the following OpenSSL badge [questions](https://github.com/kubeflow/model-registry/issues/1084#:~:text=The%20security%20mechanisms,cryptographically%20insecure.%20%5Bcrypto_random%5D) (the currently unaswered ones), please?